kitsune/sumo/api_utils.py

```diff
@@ -298,51 +253,35 @@ def has_object_permission(self, request, view, obj):
         return user == creator
 
 
-def PermissionMod(cls, permissions):
+PermissionListSerializer = None
+
+
+def PermissionMod(field, permissions=[]):
     """
     Takes a class and modifies it to conditionally hide based on permissions.
     """
-    class Modded(cls):
-
-        def __init__(self, **kwargs):
-            super(Modded, self).__init__(**kwargs)
-            self._stealth = False
-
-        def field_to_native(self, obj, field_name):
-            """
-            Return null value if request has no access to that field
-            """
-            if self.check_permissions() and self.check_object_permissions(obj):
-                self._stealth = False
-                return super(Modded, self).field_to_native(obj, field_name)
-            self._stealth = True
-            return None
-
-        # write_only is converted to a property, so that it can be "stealthed."
-        # The write only field is how DRF chooses to include or not include a field
-        # in serialization, so by controlling it, this field can choose when it
-        # is included in serialization output.
 
-        @property
-        def write_only(self):
-            return self._stealth or self._write_only
+    class Modded(field):
+        @classmethod
+        def many_init(cls, *args, **kwargs):
+            kwargs['child'] = field()
+            return PermissionMod(serializers.ListSerializer, permissions)(*args, **kwargs)
 
-        @write_only.setter
-        def write_only(self, value):
-            self._write_only = value
-
-        # These methods mimic the methods found on DRF views.
-
-        def get_permissions(self):
-            return (p() for p in permissions)
-
-        def check_permissions(self):
-            request = self.context.get('request', None)
-            return all(p.has_permission(request, self) for p in self.get_permissions())
-
-        def check_object_permissions(self, obj):
-            request = self.context.get('request', None)
-            return all(p.has_object_permission(request, self, obj) for p in self.get_permissions())
+        def get_attribute(self, instance):
+            if self.check_permissions(instance):
+                return super(Modded, self).get_attribute(instance)
+            else:
+                raise fields.SkipField()
+
+        def check_permissions(self, obj):
+            request = self.context.get('request')
+            for Perm in permissions:
+                perm = Perm()
+                if not perm.has_permission(request, self):
+                    return False
+                if not perm.has_object_permission(request, self, obj):
+                    return False
+            return True
 
     return Modded
```

Comment:
I think this messes up introspection of this thing. With functions you have `@wraps`. Classes are a little harder. This might be useful, but I only skimmed it:

http://stackoverflow.com/questions/6394511/python-functools-wraps-equivalent-for-classes

I don't know if it matters a ton, but it might if you were debugging something that involved this.